hs/src/Reach/Backend/JS.hs

```diff
@@ -879,8 +879,44 @@ jsMapDefns varsHuh = do
 jsError :: Doc -> Doc
 jsError err = "new Error(" <> err <> ")"
 
-jsPart :: DLInit -> SLPart -> EPProg -> App Doc
-jsPart dli p (EPProg _ ctxt_isAPI _ et) = do
+setupPart :: Doc -> [Doc]
+setupPart who = [
+  ctcTopChk who
+  , interactChk who
+  , "const ctc = ctcTop._initialize();"
+  , "const stdlib = ctc.stdlib;" ]
+
+jsApiWrapper :: B.ByteString -> [Int] -> App Doc
+jsApiWrapper p whichs = do
+  let who = pretty $ bunpack p
+  let chk_which w = "step == " <> pretty w
+  let chk_st = concatWith (\ l r -> l <> " || " <> r) $ map chk_which whichs
+  let assertStep = "stdlib.assert" <> parens (chk_st <> ", 'API called in the wrong state. Currently in state: ' + step ") <> semi
+  let jmps = map (\ which -> do
+          let inst = "_" <> who <> pretty which
+          "if" <+> parens ("step" <+> "==" <+> pretty which) <+> braces ("return " <> inst <> parens "ctcTop, interact" <> semi)
+        ) whichs
+  let body = vsep $
+        setupPart who
+        <> [ "const step = await ctc.getCurrentStep()"
```

Comment:
This function `getCurrentStep` should go into the `interfaces.ts` and you should increment the `backendVersion` in the `JS.hs` file

Reply:
And make a note in the changelog

Reply:
I thought methods on a `Contract` get annotated in `shared_impl` and only top level things from the runtime are in `interfaces`?

Reply:
https://github.com/reach-sh/reach-lang/blob/master/hs/src/Reach/Backend/JS.hs#L1073 is used to ensure that the compiled object's expectations match the things provided by the stdlib and we don't get weird undefined errors. It has to match here --- https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/ALGO.ts#L159

I pointed to the wrong file. You need to add it to `IContractCompiled` --- https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/shared_impl.ts#L179